background.js

```diff
@@ -0,0 +1,108 @@
+var urls = {};
+var currentUrl = null;
+var currentDay = null;
+var startTime = null;
+const MAX_TOP_URLS = 7;
+
+function getTopUrls() {
+    updateCurrentTime();
+	var items = sortUrls(urls);
+    return items.slice(0, Math.Min(MAX_TOP_URLS, items.length));
+}
+
+function sortUrls(urls) {
+    if (urls == null)
+        return [];
+
+	// Create items array
+	var items = Object.keys(urls).map(function(key) {
+		return [key, urls[key]];
+	});
+
+	// Sort the array based on the second element
+	items.sort(function(first, second) {
+		return second[1] - first[1];
+	});
+
+    return items;
+}
+
+function resetDay(d) {
+    var today = d.getDay();
+
+    if (currentDay == null) {
+        currentDay = today;
+        return;
+    }
+
+    if (currentDay != today) {
+        alert("reset");
+        urls = {}
+        currentUrl = null;
+        startTime = null;
+        currentDay = today;
+    }
+}
+
+function formatUrl(url) {
```

Comment:
Instead of manual string parsing, use this:
https://gist.github.com/jlong/2428561